app/controllers/admin/orders_controller.rb

```diff
@@ -0,0 +1,50 @@
+class Admin::OrdersController < ApplicationController
+  before_action :require_admin
+
+  def show
+    @cart_items = cart.items
+    @order = Order.find(params[:id])
+    items = []
+    @order.order_items.each do |item|
+      item.quantity.times {items << Item.find(item.item_id)}
+    end
+     @items = items.group_by(&:itself)
+  end
+
+  def index
+    type_of_order
+    @cart_items = cart.items
+  end
+
+  def update
+    @order = Order.find(params[:id])
+    order_status
+    redirect_to admin_order_path(@order.id)
+  end
+
+private
+
+  def order_status
+    if params[:commit]    == "Mark as Paid"
+      @order.status        = "Paid"
+    elsif params[:commit] == "Cancel"
+      @order.status        = "Canceled"
+    else params[:commit]  == "Completed"
+      @order.status        = "Completed"
+    end
+    @order.save
+  end
```

Comment:
There's a lot going on here. Can you think of ways to refactor so we can get rid of the if/else statements? A hash would be one way where they keys are the commit params and then the values are the status you want to update with. An example, `order_status = {"Mark as Paid" => "Paid", "Cancel" => "Canceled"...}`, then you could say `@order.status = order_status[params[:commit]]` Another thing you could do would be to have a method on `Order` like `paid!` which would do `@order.status = "Paid"` instead of doing the logic in the controller.